config.js

```diff
@@ -238,114 +238,130 @@ exports.updateFromArgv = function() {
-    }
-
-    if (argv.maxmaintime || argv.maxmaintimeranked || argv.maxmaintimeunranked || argv.minmaintime || argv.minmaintimeranked || argv.minmaintimeunranked) {
-        console.log("Warning: --min/max*maintime*+/-ranked/unranked is not supported anymore\n Use --min/max*maintime*blitz/live/corr*+/-ranked/unranked");
-    }
-
-    if (argv.maxperiodtime || argv.maxperiodtimeranked || argv.maxperiodtimeunranked || argv.minperiodtime || argv.minperiodtimeranked || argv.minperiodtimeunranked) {
-        console.log("Warning: --min/max*periodtime*+/-ranked/unranked is not supported anymore\n Use --min/max*periodtime*blitz/live/corr*+/-ranked/unranked");
-    }
-
-    if (argv.maxperiods && (argv.maxperiodsranked || argv.maxperiodsunranked)) {
-        console.log("Warning: You are using --maxperiods in combination with --maxperiodsranked and/or --maxperiodsunranked.\nUse either --maxperiods alone, OR --maxperiodsranked with --maxperiodsunranked.\nBut don't use the 3 maxperiods arguments at the same time.");
-    }
-
-    if (argv.minperiods && (argv.minperiodsranked || argv.minperiodsunranked)) {
-        console.log("Warning: You are using --minperiods in combination with --minperiodsranked and/or --minperiodsunranked.\nUse either --minperiods alone, OR --minperiodsranked with --minperiodsunranked.\nBut don't use the 3 minperiods arguments at the same time.");
-    }
+    /*** console : DEPRECIATIONS ***/
+    console.log("-----DEPRECIATIONS-----");
+    this.indicatorConsoleDepreciation = false;
 
-    if (argv.minrank && (argv.minrankranked || argv.minrankunranked)) {
-        console.log("Warning: You are using --minrank in combination with --minrankranked and/or --minrankunranked. \n Use either --minrank alone, OR --minrankranked with --minrankunranked.\nBut don't use the 3 minrank arguments at the same time.");
-    }
-
-    if (argv.maxrank && (argv.maxrankranked || argv.maxrankunranked)) {
-        console.log("Warning: You are using --maxrank in combination with --maxrankranked and/or --maxrankunranked. \n Use either --maxrank alone, OR --maxrankranked with --maxrankunranked.\nBut don't use the 3 maxrank arguments at the same time.");
-    }
-
-    if (argv.nopause && (argv.nopauseranked || argv.nopauseunranked)) {
-        console.log("Warning: You are using --nopause in combination with --nopauseranked and/or --nopauseunranked. \n Use either --nopause alone, OR --nopauseranked with --nopauseunranked.\nBut don't use the 3 nopause arguments at the same time.");
-    }
-
-    if (argv.komi) {
-        if (argv.komi.includes(`auto`)) {
-            console.log("Warning: /--komi auto/ has been renamed to /--komi automatic/\n");
-            // we skip a line here, not below, because this argv may be undefined if not used by bot admin
+    function depreciatedArgumentIsUsedNewArgumentToUseInstead(depreciatedArgumentUsed, newArgumentToUseInstead, indicatorDepreciation) {
```

Comment:
I'm pretty sure passing in `indicatorDepreciation` here does not work as you intend it to work:

```
let a = false;
function b(a) {
	a = true;
}
b(a);
display.log(a);
```
Results in: `false`

On the other hand, having the function refer to the same variable as captured from the scope of the function definition:

```
let a = false;
function b() {
	a = true;
}
b();
display.log(a);
````

Results in: `true`.

Read about closures if this sounds unexpected, for example [this](https://www.w3schools.com/js/js_function_closures.asp).

Reply:
i'm starting to understand it a little

so there are many things to notice, but the first thing i notice is you should put parameters in a function only if you want to process and modify that parameter

for example 

```
function f(x) {
    x = x*x;
    console.log(x);
}
```
is wrong, but then : 

```
function f(x) {
    return x*x;
}
```
will return the processed value, i see !!
very interesting :+1:

Reply:
so !!

if i am understanding it correctly, :1st_place_medal: 

then it means i should first convert the wanted arg to a universal one, then apply universal function for it, same as in the UHMAEATs

```
let universalArgument = "";
universalArgument = config.maxtotalgames;
deprecatedUniversalArgumentCheck();
universalArgument = config.maxtotalgames;
deprecatedUniversalArgumentCheck();
```

@windo i seee !!
very clear now !!

Reply:
just added OOP now :+1:

Reply:
and actually i can use an array instead for faster code !!